src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -841,8 +841,12 @@ public bool Equals(Guid g)
             // Now compare each of the elements
             return g._a == _a &&
                 Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
+#if BIT64
```

Comment:
Why not to do this for the first check as well?

Reply:
@jkotas in case if two guids are different (both random) - it becomes slower (see `Equals_2` in the table - it does two ulong-checks).

Reply:
* changed to be two ulong *